.golangci.yml

```diff
@@ -44,7 +45,9 @@ issues:
     - linters:
         - godot
       source: "^// ==="
-
+    - linters:
+        - perfsprint
+      text: "fmt.Sprintf can be replaced with string addition"
```

Comment:
Is that really the right text? It seems you are even excluding `strconcat` below.

Reply:
Yes, it's. These lines ignores the following `perfsprint` issues to make this PR a little shorter for review:

<details>
<summary>Details</summary>

```
❯ ./bin/golangci-lint run
cmd/prometheus/main.go:114:24: fmt.Sprintf can be replaced with string addition (perfsprint)
        return app.Flag(name, fmt.Sprintf("%s Use with server mode only.", help)).
                              ^
cmd/prometheus/main.go:124:24: fmt.Sprintf can be replaced with string addition (perfsprint)
        return app.Flag(name, fmt.Sprintf("%s Use with agent mode only.", help)).
                              ^
cmd/prometheus/query_log_test.go:63:30: fmt.Sprintf can be replaced with string addition (perfsprint)
        l, err := net.Listen("tcp", fmt.Sprintf("%s:0", p.host))
                                    ^
discovery/azure/azure.go:74:14: fmt.Sprintf can be replaced with string addition (perfsprint)
        userAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                    ^
discovery/digitalocean/digitalocean.go:134:21: fmt.Sprintf can be replaced with string addition (perfsprint)
                godo.SetUserAgent(fmt.Sprintf("Prometheus/%s", version.Version)),
                                  ^
discovery/digitalocean/mock_test.go:59:39: fmt.Sprintf can be replaced with string addition (perfsprint)
                if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", tokenID) {
                                                    ^
discovery/eureka/client.go:26:17: fmt.Sprintf can be replaced with string addition (perfsprint)
var userAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                ^
discovery/hetzner/mock_test.go:59:39: fmt.Sprintf can be replaced with string addition (perfsprint)
                if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", hcloudTestToken) {
                                                    ^
discovery/hetzner/mock_test.go:435:39: fmt.Sprintf can be replaced with string addition (perfsprint)
                if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", hcloudTestToken) {
                                                    ^
discovery/hetzner/robot.go:42:17: fmt.Sprintf can be replaced with string addition (perfsprint)
var userAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                ^
discovery/http/http.go:46:21: fmt.Sprintf can be replaced with string addition (perfsprint)
        userAgent        = fmt.Sprintf("Prometheus/%s", version.Version)
                           ^
discovery/ionos/server.go:80:18: fmt.Sprintf can be replaced with string addition (perfsprint)
        cfg.UserAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                        ^
discovery/ionos/server_test.go:101:17: fmt.Sprintf can be replaced with string addition (perfsprint)
                http.Error(w, fmt.Sprintf("bad url: %s", r.URL.Path), http.StatusNotFound)
                              ^
discovery/linode/linode.go:168:22: fmt.Sprintf can be replaced with string addition (perfsprint)
        client.SetUserAgent(fmt.Sprintf("Prometheus/%s", version.Version))
                            ^
discovery/linode/mock_test.go:59:40: fmt.Sprintf can be replaced with string addition (perfsprint)
                        if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", tokenID) {
                                                            ^
discovery/moby/dockerswarm.go:40:17: fmt.Sprintf can be replaced with string addition (perfsprint)
var userAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                ^
discovery/puppetdb/puppetdb.go:65:21: fmt.Sprintf can be replaced with string addition (perfsprint)
        userAgent        = fmt.Sprintf("Prometheus/%s", version.Version)
                           ^
discovery/scaleway/baremetal.go:96:21: fmt.Sprintf can be replaced with string addition (perfsprint)
                scw.WithUserAgent(fmt.Sprintf("Prometheus/%s", version.Version)),
                                  ^
discovery/scaleway/instance.go:107:21: fmt.Sprintf can be replaced with string addition (perfsprint)
                scw.WithUserAgent(fmt.Sprintf("Prometheus/%s", version.Version)),
                                  ^
discovery/vultr/mock_test.go:58:39: fmt.Sprintf can be replaced with string addition (perfsprint)
                if r.Header.Get("Authorization") != fmt.Sprintf("Bearer %s", APIKey) {
                                                    ^
discovery/vultr/vultr.go:137:24: fmt.Sprintf can be replaced with string addition (perfsprint)
        d.client.SetUserAgent(fmt.Sprintf("Prometheus/%s", version.Version))
                              ^
discovery/xds/client.go:33:17: fmt.Sprintf can be replaced with string addition (perfsprint)
var userAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                ^
discovery/xds/client.go:143:70: fmt.Sprintf can be replaced with string addition (perfsprint)
        serverURL.Path = path.Join(serverURL.Path, string(protocolVersion), fmt.Sprintf("discovery:%s", resourceType))
                                                                            ^
notifier/notifier.go:56:17: fmt.Sprintf can be replaced with string addition (perfsprint)
var userAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                ^
promql/bench_test.go:425:11: fmt.Sprintf can be replaced with string addition (perfsprint)
                name := fmt.Sprintf("%s (should fail)", c)
                        ^
promql/engine.go:103:9: fmt.Sprintf can be replaced with string addition (perfsprint)
        return fmt.Sprintf("query timed out in %s", string(e))
               ^
promql/engine.go:107:9: fmt.Sprintf can be replaced with string addition (perfsprint)
        return fmt.Sprintf("query was canceled in %s", string(e))
               ^
promql/engine.go:111:9: fmt.Sprintf can be replaced with string addition (perfsprint)
        return fmt.Sprintf("query processing would load too many samples into memory in %s", string(e))
               ^
promql/engine_test.go:4464:19: fmt.Sprintf can be replaced with string addition (perfsprint)
                                queryString = fmt.Sprintf(`%s{idx="0"}`, seriesName)
                                              ^
promql/engine_test.go:4718:20: fmt.Sprintf can be replaced with string addition (perfsprint)
                                queryString := fmt.Sprintf(`%s{idx="0"}`, seriesName)
                                               ^
promql/parser/prettier.go:61:8: fmt.Sprintf can be replaced with string addition (perfsprint)
                s += fmt.Sprintf("%s,\n", e.Param.Pretty(level+1))
                     ^
promql/parser/prettier.go:100:8: fmt.Sprintf can be replaced with string addition (perfsprint)
                s += fmt.Sprintf("%s,\n", e[i].Pretty(level))
                     ^
scrape/scrape.go:726:17: fmt.Sprintf can be replaced with string addition (perfsprint)
var UserAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                ^
storage/remote/client.go:44:17: fmt.Sprintf can be replaced with string addition (perfsprint)
var UserAgent = fmt.Sprintf("Prometheus/%s", version.Version)
                ^
storage/remote/codec_test.go:442:65: fmt.Sprintf can be replaced with string addition (perfsprint)
        require.Equal(t, "duplicate label with name: foo", errMessage, fmt.Sprintf("Expected error to be from duplicate label, but got: %s", errMessage))
                                                                       ^
tsdb/chunks/chunks.go:195:10: fmt.Sprintf can be replaced with string addition (perfsprint)
                        panic(fmt.Sprintf("unknown sample type %s", sampleType.String()))
                              ^
tsdb/compact_test.go:1588:15: fmt.Sprintf can be replaced with string addition (perfsprint)
                                                        "job", fmt.Sprintf("sparse_histogram_schema_%s", schemaDescription[sid]),
                                                               ^
tsdb/db_test.go:4958:9: fmt.Sprintf can be replaced with string addition (perfsprint)
                t.Run(fmt.Sprintf("name=%s", tc.name), func(t *testing.T) {
                      ^
tsdb/db_test.go:5042:9: fmt.Sprintf can be replaced with string addition (perfsprint)
                t.Run(fmt.Sprintf("name=%s", tc.name), func(t *testing.T) {
                      ^
tsdb/head_read_test.go:132:9: fmt.Sprintf can be replaced with string addition (perfsprint)
                t.Run(fmt.Sprintf("name=%s", tc.name), func(t *testing.T) {
                      ^
tsdb/ooo_head_read_test.go:831:9: fmt.Sprintf can be replaced with string addition (perfsprint)
                t.Run(fmt.Sprintf("name=%s", tc.name), func(t *testing.T) {
                      ^
tsdb/ooo_head_read_test.go:996:9: fmt.Sprintf can be replaced with string addition (perfsprint)
                t.Run(fmt.Sprintf("name=%s", tc.name), func(t *testing.T) {
                      ^
tsdb/ooo_head_read_test.go:1127:9: fmt.Sprintf can be replaced with string addition (perfsprint)
                t.Run(fmt.Sprintf("name=%s", tc.name), func(t *testing.T) {
                      ^
tsdb/ooo_head_read_test.go:1191:9: fmt.Sprintf can be replaced with string addition (perfsprint)
                t.Run(fmt.Sprintf("name=%s", tc.name), func(t *testing.T) {
                      ^
util/documentcli/documentcli.go:97:34: fmt.Sprintf can be replaced with string addition (perfsprint)
        return writeTable(writer, rows, fmt.Sprintf("%s Flags", strings.Repeat("#", level+2)))
                                        ^
util/documentcli/documentcli.go:128:34: fmt.Sprintf can be replaced with string addition (perfsprint)
        return writeTable(writer, rows, fmt.Sprintf("%s Arguments", strings.Repeat("#", level+2)))
                                        ^
web/api/v1/api_test.go:2870:32: fmt.Sprintf can be replaced with string addition (perfsprint)
                r, err := http.NewRequest(m, fmt.Sprintf("http://example.com?%s", q.Encode()), nil)
                                             ^
web/api/v1/api_test.go:3175:43: fmt.Sprintf can be replaced with string addition (perfsprint)
                        req, err := http.NewRequest(tc.method, fmt.Sprintf("?%s", tc.values.Encode()), nil)
                                                               ^
web/api/v1/api_test.go:3551:43: fmt.Sprintf can be replaced with string addition (perfsprint)
                        req, err := http.NewRequest(tc.method, fmt.Sprintf("?%s", tc.values.Encode()), nil)
                                                               ^
web/api/v1/api_test.go:3881:43: fmt.Sprintf can be replaced with string addition (perfsprint)
                        req, err := http.NewRequest(tc.method, fmt.Sprintf("http://example.com?%s", query.Encode()), nil)
                                                               ^
```

</details>

I removed `strconcat` because this option appeared only in golangci-lint v1.57.1. See https://github.com/golangci/golangci-lint/blob/v1.57.1/.golangci.reference.yml#L1448-L1450

Reply:
Do I understand correctly that v1.56.2 creates lint errors for the string concatenation case, but doesn't have an option to disable them, which in turn is the reason you exclude them based on the error text?

Reply:
@beorn7 yes. The issues excluded by text because it's not possible to exclude by option in golangci-lint v1.56.2.

Reply:
I see this PR does move 2 Sprintf to concatenation (`Source: "OS_" + h.region,`), I think it's intentional (or a leftover).
You're welcome to upgrade `golangci-lint` if no PR is there ;)